### PR TITLE
Various aesthetic/usability improvements to frontend

### PR DESCRIPTION
**Pull request by pbotros**
General:
- Support for switching between tabs (e.g. "Editor" and "Simulator")
  without resetting simulator buttons / state. Added a "Re-assemble"
  button for students to click when they've edited things and want to
  re-assemble.

Simulator tab:
- Support for displaying breakpoints in simulator with dot (helps when
  the current instruction has a breakpoint and would otherwise be
  invisible)
- Less harsh colors: copied the blue/reds from JetBrains theme
- Modern fonts: copied monospace font ordering from bulma
- Support for proper search bar in the "Address" tab: can click Go or
  press enter now
- Only force-scroll the simulator instructions if the instruction is out
  of view
- More compact representation for registers

Some screenshots:
venus_dark_mode_review
venus_light_mode_review
venus_address_search

@jerowang PTAL? Lmk if a GIF or more screenshots would be helpful, or else feel free to try it out on your own locally too.

### CODE REVIEW

File: src/main/kotlin/venus/Renderer.kt [hunk 1]
```diff
@@ -95,16 +95,20 @@ internal object Renderer {
 
     fun renderSimButtons() {
         val simbtns = document.getElementById("simulator-buttons") as HTMLDivElement
-        val simassmbbtns = document.getElementById("simulator-assemble-buttons") as HTMLDivElement
-        simassmbbtns.style.display = "none"
+        // Turn all simulator-related buttons on
```
Comment (jerowang) [inline src/main/kotlin/venus/Renderer.kt]: can this comment be moved up/down one line and same with the comment on line 107

File: src/main/kotlin/venus/Renderer.kt [hunk 2]
```diff
@@ -244,20 +248,30 @@ internal object Renderer {
         newRow.id = "instruction-$pcx"
         newRow.onclick = { Driver.addBreakpoint(pcx) }
 
-        val pcline = newRow.insertCell(0)
+        var cellIndex = 0
+        val breakpointLine = newRow.insertCell(cellIndex)
+        val breakpointNode = document.createElement("div")
+        breakpointNode.setAttribute("name", "breakpoint_cell")
+        breakpointLine.appendChild(breakpointNode)
+        cellIndex += 1
```
Comment (jerowang) [inline src/main/kotlin/venus/Renderer.kt]: `++` please, I think this repo mostly uses it (same with line 261)

**Review by jerowang — APPROVED** [review]
lgtm, as Jerry said, don't merge yet please

File: src/main/kotlin/venus/Renderer.kt (hunk 1 above)
Comment (pbotros) [reply src/main/kotlin/venus/Renderer.kt]: Done

File: src/main/kotlin/venus/Renderer.kt (hunk 2 above)
Comment (pbotros) [reply src/main/kotlin/venus/Renderer.kt]: Done

**Comment by pbotros** [comment]
Thanks @jerowang - comments addressed, will hold off on merging.

**Pull request merged**